rpc/client/client.go

```diff
@@ -361,9 +363,70 @@ func (c *Client) Tx(ctx context.Context, hash []byte, prove bool) (*ctypes.Resul
 	panic("Tx - not implemented!")
 }
 
-func (c *Client) TxSearch(ctx context.Context, query string, prove bool, page, perPage *int, orderBy string) (*ctypes.ResultTxSearch, error) {
-	// needs block store
-	panic("TxSearch - not implemented!")
+func (c *Client) TxSearch(ctx context.Context, query string, prove bool, pagePtr, perPagePtr *int, orderBy string) (*ctypes.ResultTxSearch, error) {
+	q, err := tmquery.New(query)
+	if err != nil {
+		return nil, err
+	}
+
+	results, err := c.node.TxIndexer.Search(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	// sort results (must be done before pagination)
+	switch orderBy {
+	case "desc":
+		sort.Slice(results, func(i, j int) bool {
+			if results[i].Height == results[j].Height {
+				return results[i].Index > results[j].Index
+			}
+			return results[i].Height > results[j].Height
+		})
+	case "asc", "":
+		sort.Slice(results, func(i, j int) bool {
+			if results[i].Height == results[j].Height {
+				return results[i].Index < results[j].Index
+			}
+			return results[i].Height < results[j].Height
+		})
+	default:
+		return nil, errors.New("expected order_by to be either `asc` or `desc` or empty")
+	}
+
+	// paginate results
+	totalCount := len(results)
+	perPage := validatePerPage(perPagePtr)
+
+	page, err := validatePage(pagePtr, perPage, totalCount)
+	if err != nil {
+		return nil, err
+	}
+
+	skipCount := validateSkipCount(page, perPage)
+	pageSize := tmmath.MinInt(perPage, totalCount-skipCount)
+
+	apiResults := make([]*ctypes.ResultTx, 0, pageSize)
+	for i := skipCount; i < skipCount+pageSize; i++ {
+		r := results[i]
+
+		var proof types.TxProof
+		/*if prove {
+			block := nil                               //env.BlockStore.LoadBlock(r.Height)
+			proof = block.Data.Txs.Proof(int(r.Index)) // XXX: overflow on 32-bit machines
+		}*/
```

Comment:
what's the implications of this?

Reply:
The response for this Tx search would always be without an inclusion proof for the resulting Txs. 

Completely orthogonal to this but now that I see this: @evan-forbes I think we have never updated `Txs.Proof` to match to the data root (mix split into shares, NMT for daheader + binary tree for data root) we have instead of the [plain binary Merkle tree tendermint uses](https://github.com/celestiaorg/celestia-core/blob/3f08c522c7dcde720dca86a714f5d114043da779/types/tx.go#L74) 🤔 Do we have an issue for that? We might have decided that this could also be served by the celestia-node's rpc instead but I'm not sure we are tracking this there either. Use-cases like this where e.g. EVMOS indirectly uses that end-point wouldn't be a problem as Celestia-core is not meant to be used to build stuff like this on top. But we should still be aware of this.

Reply:
I removed it just to get things to run, but none of the eth RPC endpoints for Get*Tx never have inclusion proofs (i just checked), so i don't it matters for evmos.

Reply:
Makes sense, thanks for clarifying.